week02/valid_paranthesis.cpp

```diff
@@ -0,0 +1,42 @@
+class Solution {
+public:
+    bool isValid(string s) {
+        stack<char>paranStack;
+        int size = s.size();
+        
+        for(int i = 0; i < size; i++){
+            
+            if(s[i] == '(' || s[i] == '{' || s[i] == '['){
+                paranStack.push(s[i]);
+            }
+            else{
+                if(paranStack.empty()){
+                    return false;
+                }
+              
+                if(s[i] == ')' && paranStack.top() == '('){
+                     paranStack.pop();
+                 }
+                else if(s[i] == '}' && paranStack.top() == '{'){
+                    paranStack.pop();
+                }
+                else if(s[i] == ']' && paranStack.top() == '['){ 
+                        paranStack.pop();
+                }
```

Comment:
It can be optimized by using `unordered_map`. Kindly refer to our [editorial](https://docs.google.com/document/d/1VipVcGyqiUDvFWbJ_yivTJucBRw3ct7HGZGGx-_CRsk/edit).